### DISCUSSION TITLE
My header and footer are getting inverted on adding them to a PDFs that I merged from multiple PDFs.

### DISCUSSION THREAD

**Discussion by Harshitttttttt** [pymupdf/PyMuPDF#3352] [Looking for help] [answered]
If I add the header and footer on a different document, it works properly but on my merged document the text is inverted.

This is the code I am using to merge my PDFs
```
def merge_pdfs(pdf_dir, output_filename, number_of_pdfs, format_of_file_name):
    merged_doc = fitz.open()
    page_count_list = []
    for idx in range(1, number_of_pdfs+1):
        filename = format_of_file_name.replace('i', str(idx)) + ".pdf"
        filepath = os.path.join(pdf_dir, filename)
        if os.path.isfile(filepath):
            src_doc = fitz.open(filepath)
            merged_doc.insert_pdf(src_doc)
            page_count_list.append(src_doc.page_count)
            src_doc.close()
        else:
            print(f"File {filename} not found.")
            
    merged_doc.save(output_filename)
```

and then I access the same file to add the header and the footer
```
def add_header_or_footer(output_filename):
    temp_output_filename = output_filename.replace(".pdf", "_h&f.pdf")
    doc = fitz.open(output_filename)
    header = input("Enter the text for the header: ")
    footer = input("Enter the text for the footer: ")
    for page in doc:
        page.insert_text((page.rect.width/2, 50), header)  # insert header
        page.insert_text((page.rect.width-75-len(footer), page.rect.height - 50), footer) # insert footer
        
    doc.save(temp_output_filename)
    print(f"Header and footer added to {output_filename}")
```

but the problem is the header and footer are both inverted
![image](https://github.com/pymupdf/PyMuPDF/assets/106076610/fd039a88-9a21-4699-b71a-35687ed2371d)
![image](https://github.com/pymupdf/PyMuPDF/assets/106076610/e33a0cdc-046b-49e0-8a6f-2aa43533a700)
![image](https://github.com/pymupdf/PyMuPDF/assets/106076610/a391ed58-4442-40cc-8353-c61914bdb39f)

Any help would be appreciated and it was fun learning PyMuPDF thanks!

**Comment by JorjMcKie** [comment, marked as answer]
Immediately after loading a page execute `page.clean_contents()`. This should solve the problem. See also [here](https://pymupdf.readthedocs.io/en/latest/recipes-common-issues-and-their-solutions.html#misplaced-item-insertions-on-pdf-pages).

**Reply by Harshitttttttt** [reply]
It worked, thanks a lot!
